doc/transactions.md

```diff
@@ -0,0 +1,78 @@
+Transactions
+============
+Transactions is a feature that comes from SOFT, which allows to easy manage arbitrary long series of immutable (frozen) instances while ensuring provenance.  Conceptually it share many similarities with git.
+
+A basic usage for transactions is to take snapshots of the current state of your system.  This is shown in Figure 1, where we assume that the instance `A` (which e.g. could be a collection) describes the state of your system.
+
+We call an instance a *transaction* after we have taken a snapshot of it, i.e. when it contains a reference to a immutable "parent" instance.
+
+![transactions](figs/transactions.png)
+
+**Figure 1**: *Creating a transaction by taking snapshots.  (a) initial state of your system described by instance `A`.  (b) creating a transaction by taking a snapshot
+of `A`.  The snapshot, `A1`, stores an immutable (frozen) copy of `A` from the exact
+moment `t1` the snapshot was taken.  (c) After another snapshot was taken at time `t2`.  Blue circles represent immutable instances, red circles represent mutable instances, while arrows relate an instance to its (frozen) parent instance.*
+
+It is very simple to create such snapshots using the Python and C APIs.  In Python, the steps in Figure 1 can be produced by the following code:
+
+```python
+A = dlite.get_instance("A")  # Initial state (Fig. 1a)
+A.snapshot()                 # Make snapshot at time=t1 (Fig. 1b)
+# A is evolved...
+A.snapshot()                 # Make snapshot at time=t2 (Fig. 1c)
+```
+
+The corresponding code in C would be:
+
+```C
+DLiteInstance *A = dlite_instance_get("A");  // Initial state (Fig. 1a)
+dlite_instance_snapshot(A);                  // Make snapshot at time=t1 (Fig. 1b)
+// A is evolved...
+dlite_instance_snapshot(A);                  // Make snapshot at time=t2 (Fig. 1c)
+```
+
+The snapshots can be accessed using `dlite_instance_get_snapshot()`.  For instance, accessing snapshot `A2` and creating a new branch from it can be achieved by the following four lines of Python C code:
```

Comment:
```suggestion
The snapshots can be accessed using `dlite_instance_get_snapshot()`.
For instance, accessing snapshot `A2` and creating a new branch from it can be achieved by the following four lines of Python code:
```